### PR TITLE
Rework dockerization

### LINKED ISSUE #2: Use absolute paths everywhere
That's what @jackfirth says I should do.

### LINKED ISSUE #3: Use installation scope
use installation scope everywhere.

### LINKED ISSUE #4: Make base image configurable via ARG
Allows easily building on non-ubuntu base images, or ubuntu images with different versions. The dockerfile docs have an [example](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact).

### LINKED ISSUE #5: Isolate test commands and build commands from each other
Generally, `docker build` **should not** run tests; tests should only be run when `docker run` is used on the built image. During an image build, each individual `RUN` command runs in a new build-time container and the resulting filesystem changes are committed as a diff, then that build-time container is destroyed. This defeats some assumptions of various tools, for instance `xvfb-run` starts a background process that it intends to reuse across invocations of `xvfb-run`. But because each RUN command happens in a fresh container, that doesn't work and results in a lot of waste.

Test commands should be put into some `test.sh` script that is `COPY`'d into the image, then a `CMD` instruction can specify that the `test.sh` script should be run by default whenever a container is started from the image.

### LINKED ISSUE #8: Configure builds and tests with docker compose
This allows building and running multiple images and containers at once with commands like `docker-compose build` and `docker-compose run`.

### LINKED ISSUE #9: Replace `/usr/bin/racket -e '(+ 1 2)'` test with `docker run` test
Given an image with Racket installed, `docker run <image> racket -e '(+ 1 2)'` will check that a container spawned from the image can evaluate Racket code. This test can itself be in a docker container using docker-in-docker with the host docker socket bind-mounted, making it easy to integrate with docker-compose.

### LINKED ISSUE #11: Keep intermediate stuff out of final images with multi-stage builds
Specifically, `apt-get` downloads a bunch of stuff and `wget` is only needed to download the installer. Using a tiny Alpine Linux based image for the `wget` stage (such as [mwendler/wget](https://hub.docker.com/r/mwendler/wget/)) would be best.

### LINKED ISSUE #14: Test i386 installers

### PR DESCRIPTION
**Pull request by jackfirth**
This PR changes how Docker is used to be more in line with Docker's philosophical assumptions and best practices. More pragmatically, it's now much easier to run the release tests for _every_ installer variant, not just one. In-depth change descriptions can be found in each commit message.

Closes #2 
Closes #3 (I think)
Closes #4 
Closes #5 
Closes #8 
Closes #9 
Closes #11 
Closes #14 

Copyright Google, see first commit for details.

### CODE REVIEW

File: Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,28 @@
+ARG RACKET_IMAGE
+
+FROM ${RACKET_IMAGE}
+
+RUN raco pkg install --auto --installation \
```
Comment (samth) [inline Dockerfile.test]: I think we should just install `main-distribution-test` instead of most of these.

**Review by samth — APPROVED** [review]
Woo!

**Comment by samth** [comment]
Actually, can you also fix the readme so that I know what to run?

**Comment by samth** [comment]
Ok, there are a few things missing from this:

1. Builds with `-funsigned-char` and `-DALTERNATE_TARGET_REGISTER`
2. Compiling from source for both the minimal and regular distributions.

I think that's it that this misses from the code I've written.

File: Dockerfile.test (hunk 1 above)
Comment (jackfirth) [reply Dockerfile.test]: Done, I think.

**Comment by jackfirth** [comment]
Would you prefer we add those cases before merging this PR or after?

**Comment by samth** [comment]
We can merge this PR first. There's a while before the next release cycle.

**Comment by samth** [comment]
I think we can merge this, and start working on further improvements.

**Pull request merged**